dist/api_data.py

```diff
@@ -1557,6 +1567,23 @@ def __ge__(self, other):
             'oldest','oldest_reader','pinned','recovery','stable']),
 ]),
 
+'WT_CONNECTION.query_timestamp_numeric' : Method([
+    Config('get', 'all_committed', r'''
+        specify which timestamp to query: \c all_committed returns the largest
+        timestamp such that all timestamps up to that value have committed,
+        \c last_checkpoint returns the timestamp of the most recent stable
+        checkpoint, \c oldest returns the most recent \c oldest_timestamp set
+        with WT_CONNECTION::set_timestamp, \c oldest_reader returns the
+        minimum of the read timestamps of all active readers \c pinned returns
+        the minimum of the \c oldest_timestamp and the read timestamps of all
+        active readers, \c recovery returns the timestamp of the most recent
+        stable checkpoint taken prior to a shutdown and \c stable returns the
+        most recent \c stable_timestamp set with WT_CONNECTION::set_timestamp.
+        See @ref transaction_timestamps''',
+        choices=['all_committed','last_checkpoint',
+            'oldest','oldest_reader','pinned','recovery','stable']),
+]),
+
 'WT_CONNECTION.set_timestamp' : Method([
```

Comment:
I thought in one of the meetings we concluded that timestamp_transaction and query_timestamp are the two APIs that are going to have numeric versions.
@agorrod , @michaelcahill, @bvpvamsikrishna do we need a numeric version of set_timestamp ?

Reply:
I don't think we need a version of `set_timestamp` - I think of this numeric API as an optimization/simplification path rather than a replacement for the current API. `WT_CONNECTION::set_timestamp` doesn't feel like it'll gain significant benefit from the new numeric API, so I'm inclined to skip it for now. If MongoDB wants it in the future we can add it as part of a different ticket.

Reply:
I would like to maintain consistency, we can either have the numeric implementation or not have the numeric implementation, but better to be consistent with both WT_CONNECTION.set_timestamp and WT_CONNECTION.query_timestamp. Current state WT_CONNECTION.query_timestamp is implemented but not WT_CONNECTION.set_timestamp.

Reply:
OK. I don't feel super strongly about it, so am OK to add a new `WT_CONNECTION::set_timestamp`. There are other APIs that accept timestamps as well and I don't think we should add numeric versions for all of those.

Reply:
I am feeling a bit ambivalent about this entire change. I mostly agree with @bvpvamsikrishna that set and query kind of go together. But I think that @dgottlieb's comment is most telling. The usage of the API of `WT_CONNECTION::set_timestamp("oldest_timestamp=N,stable_timestamp=N")` now has to become two API calls, and that introduces delay between those calls and ordering issues.

Reply:
I agree with Sue and Dan that we need to address the multiple TS issue. If we still will have cases where we have to use the text version, it doesn't make sense to optimize only the other call sites. I don't think the formatting is a serious perf issue but will admit that I don't know how much WT parsing adds. We recently added a new string formatting library to the server that makes generating this kind of string much simpler and faster.

Reply:
I'm not advocating for this change here - but does it make a difference that MongoDB doesn't currently call any API while passing in two timestamps to the same call?

Reply:
MongoDB does make calls with multiple timestamps on the WT_CONNECTION::set_timestamp method (though now that I'm seeing whats in master, we may be doing it wrong in one spot!).
On v4.0 that is correct: https://github.com/mongodb/mongo/blob/ddb5e3eb82bd8f22a384835923d7f20bfa479d24/src/mongo/db/storage/wiredtiger/wiredtiger_kv_engine.cpp#L1506
In master that I think is missing the (duplicated) values:
https://github.com/mongodb/mongo/blob/ae3df9b0cb1327e818dc45da6bb3f0a1feb7f539/src/mongo/db/storage/wiredtiger/wiredtiger_kv_engine.cpp#L1603
Another example in master: https://github.com/mongodb/mongo/blob/ae3df9b0cb1327e818dc45da6bb3f0a1feb7f539/src/mongo/db/storage/wiredtiger/wiredtiger_kv_engine.cpp#L1671